From 315ed31e6bf8ff2c73f2147dd1f8642d65dad6eb Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sun, 12 May 2019 12:38:23 -0400
Subject: [PATCH] Fix bliss init files

Change-Id: I7a63e98477f13b24e782a1c747c893649a8649d0
---
 config/common.mk | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/config/common.mk b/config/common.mk
index 91ddf3b53..955ba9642 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -62,14 +62,14 @@ PRODUCT_COPY_FILES += \
 
 # init.d support
 PRODUCT_COPY_FILES += \
-    vendor/bliss/prebuilt/common/bin/sysinit:system/bin/sysinit \
-    vendor/bliss/prebuilt/common/etc/init.d/00banner:system/etc/init.d/00banner \
-    vendor/bliss/prebuilt/common/etc/init.d/90userinit:system/etc/init.d/90userinit
+    $(LOCAL_PATH)/bin/sysinit:system/bin/sysinit \
+    $(LOCAL_PATH)/etc/init.d/00banner:system/etc/init.d/00banner \
+    $(LOCAL_PATH)/etc/init.d/90userinit:system/etc/init.d/90userinit
 
 # Init file
 PRODUCT_COPY_FILES += \
-    vendor/bliss/prebuilt/common/etc/init.bliss.rc:root/init.bliss.rc \
-    vendor/bliss/prebuilt/common/etc/init.blissupdater.rc:system/etc/init/init.blissupdater.rc
+    $(LOCAL_PATH)/etc/init.bliss.rc:root/init.bliss.rc \
+    $(LOCAL_PATH)/etc/init.blissupdater.rc:system/etc/init/init.blissupdater.rc
 
 # Bring in camera effects
 PRODUCT_COPY_FILES +=  \
-- 
2.17.1

