From 31d93214ded3da60f383b52864263a3278589453 Mon Sep 17 00:00:00 2001
From: GuneetAtwal <guneetatwal@gmail.com>
Date: Mon, 11 Jun 2018 17:09:40 +0530
Subject: [PATCH 42/70] android: support 32-bit binder in 64-bit kernel

Signed-off-by: GuneetAtwal <guneetatwal@gmail.com>
---
 drivers/android/Kconfig | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/android/Kconfig b/drivers/android/Kconfig
index 432e9ad770703..2d4b8c72ee51f 100644
--- a/drivers/android/Kconfig
+++ b/drivers/android/Kconfig
@@ -32,6 +32,19 @@ config ANDROID_BINDER_DEVICES
 	  created. Each binder device has its own context manager, and is
 	  therefore logically separated from the other devices.
 
+config ANDROID_BINDER_IPC_32BIT
+	bool "Use old (Android 4.4 and earlier) 32-bit binder API"
+	depends on ANDROID_BINDER_IPC
+	default !64BIT
+	---help---
+	  The Binder API has been changed to support both 32 and 64bit
+	  applications in a mixed environment.
+
+	  Enable this to support an old 32-bit Android user-space (v4.4 and
+	  earlier).
+
+	  Note that enabling this will break newer Android user-space.
+
 config ANDROID_BINDER_IPC_SELFTEST
 	bool "Android Binder IPC Driver Selftest"
 	depends on ANDROID_BINDER_IPC
-- 
2.17.1

