From a1a2b041d476b54773ee2378b0f989ed32682fe1 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Sun, 8 Feb 2015 23:39:05 +0800
Subject: [PATCH 13/80] x86/intel: force tsc to be reliable on Baytrail

---
 arch/x86/kernel/cpu/intel.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/kernel/cpu/intel.c b/arch/x86/kernel/cpu/intel.c
index fc3c07fe7df58..ba1445b12f9ca 100644
--- a/arch/x86/kernel/cpu/intel.c
+++ b/arch/x86/kernel/cpu/intel.c
@@ -239,6 +239,8 @@ static void early_init_intel(struct cpuinfo_x86 *c)
 	/* Penwell and Cloverview have the TSC which doesn't sleep on S3 */
 	if (c->x86 == 6) {
 		switch (c->x86_model) {
+		case 0x37:	/* ValleyView */
+			set_cpu_cap(c, X86_FEATURE_TSC_RELIABLE);
 		case 0x27:	/* Penwell */
 		case 0x35:	/* Cloverview */
 		case 0x4a:	/* Merrifield */
-- 
2.17.1

