From a86fd8519c5038e989ce8543db1580d20a128b6c Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Wed, 29 Oct 2014 23:25:57 +0800
Subject: [PATCH 06/28] init.rc: run the console shell as root

To make debugging easier.
---
 rootdir/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index b9464e7fd8..945d8c4d44 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -751,7 +751,7 @@ service console /system/bin/sh
     class core
     console
     disabled
-    user shell
+#   user shell
     group shell log readproc
     seclabel u:r:shell:s0
     setenv HOSTNAME console
-- 
2.17.1

