From 03d4411917d9895e31f12cbdadeefc5a4a69d8e1 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Tue, 12 Sep 2017 16:22:04 +0800
Subject: [PATCH] init: don't bail out even no SELinux domain defined

---
 init/service.cpp | 17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)

diff --git a/init/service.cpp b/init/service.cpp
index 300f9830d4..a1edf6356c 100644
--- a/init/service.cpp
+++ b/init/service.cpp
@@ -28,6 +28,7 @@
 #include <sys/wait.h>
 #include <termios.h>
 #include <unistd.h>
+#include <cutils/klog.h>
 
 #include <android-base/file.h>
 #include <android-base/logging.h>
@@ -90,19 +91,9 @@ static Result<std::string> ComputeContextFromExecutable(const std::string& servi
         free(new_con);
     }
     if (rc == 0 && computed_context == mycon.get()) {
-        if (selinux_status_getenforce() > 0) {
-            return Error() << "File " << service_path << "(labeled \"" << filecon.get()
-                           << "\") has incorrect label or no domain transition from " << mycon.get()
-                           << " to another SELinux domain defined. Have you configured your "
-                              "service correctly? https://source.android.com/security/selinux/"
-                              "device-policy#label_new_services_and_address_denials";
-        }else{
-            LOG(ERROR) << "File " << service_path << "(labeled \"" << filecon.get()
-                       << "\") has incorrect label or no domain transition from " << mycon.get()
-                       << " to another SELinux domain defined. Have you configured your "
-                          "service correctly? https://source.android.com/security/selinux/"
-                          "device-policy#label_new_services_and_address_denials";
-        }
+        KLOG_WARNING("service", "File %s (labeled \"%s\") has incorrect label or no domain transition from %s to another SELinux domain defined.",
+			service_path.c_str(), filecon.get(), mycon.get());
+		return "skip";
     }
     if (rc < 0) {
         return Error() << "Could not get process context";
-- 
2.17.1

